scripts/installation/install-openkat.yml

```diff
@@ -0,0 +1,588 @@
+  # sudo apt install postgresql
+  - name: Install other needed packages
+    apt:
+      name: "{{ item }}"
+      state: present
+    loop:
+      - python3-psycopg2
+      - python3-pexpect
+      - postgresql
+      - postgresql-client
+      - rabbitmq-server
+      - sudo
+
+  # Rocky DB
+  # Generate a secure password for the Rocky database user, as an example we’ll use /dev/urandom:
+  # echo $(tr -dc A-Za-z0-9 < /dev/urandom | head -c 20)
+  - name: Generate random password for Rocky DB
+    set_fact:
+      # Turns out that special characters create problems.
+      #var: "{{ lookup('community.general.random_string', length=20, min_numeric=3, min_upper=3, min_special=3) }}"
+      rocky_db_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+
+  - name: Show password for debugging purposes (TODO comment these lines)
+    debug:
+      var: rocky_db_password
+
+  # To configure rocky to use this password, open /etc/kat/rocky.conf and fill in this password for ROCKY_DB_PASSWORD.
+  - name: Store password in /etc/kat/rocky.conf
+    lineinfile:
+      path: /etc/kat/rocky.conf
+      regexp: "ROCKY_DB_PASSWORD=.*"
+      line:  "ROCKY_DB_PASSWORD={{ rocky_db_password }}"
+
+  # sudo -u postgres createuser rocky -P
+  - name: Create Postgresql user rocky and grant privs
+    become: true
+    become_user: postgres
+    become_method: sudo
+    postgresql_user:
+      name: rocky
+      password: "{{ rocky_db_password }}"
+      role_attr_flags: NOSUPERUSER
+
+  # sudo -u postgres createdb rocky_db
+  # sudo -u postgres psql -c 'GRANT ALL ON SCHEMA public TO rocky;' rocky_db
+  - name: Create Postgresql database rocky_db
+    become: true
+    become_user: postgres
+    become_method: sudo
+    postgresql_db:
+      name: rocky_db
+      owner: rocky
+      encoding: UTF-8
+      template: template0
+
+
+  # Now use rocky-cli to initialize the database:
+  # sudo -u kat rocky-cli migrate
+  - name: initialize Rocky database
+    become: true
+    become_user: kat
+    become_method: sudo
+    shell:
+      cmd: rocky-cli migrate
+
+  # sudo -u kat rocky-cli loaddata /usr/share/kat-rocky/OOI_database_seed.json
+  - name: Load data into OpenKAT
+    become: true
+    become_user: kat
+    become_method: sudo
+    shell:
+      cmd: rocky-cli loaddata /usr/share/kat-rocky/OOI_database_seed.json
+
+  # 
+  # The steps for creating the other databases will be similar, but we’ll explain them anyway for completeness.
+  # KAT-alogus DB
+  # 
+  # Generate a unique secure password for the KAT-alogus database user. You can use the same method we used for generating the Rocky database user password.
+  - name: Generate random password for KAT-alogus database user
+    set_fact:
+      kat_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+
+  - name: Show password for debugging purposes (TODO comment these lines)
+    debug:
+      var: kat_password
```

Comment:
Deze (en alle andere..) misschien handiger om in de vars bovenaan te zetten, netjes bij elkaar
vars:
  kat_password: "{{ lookup.... }}"
  mula_password: "{{ lookup .... }}"

Dan is het ook makkelijker aan te passen als mensen fixed passwords willen, of ze uit een eigen systeem (vault, passwordstore, etc) willen halen

Reply:
Goed idee. Pas ik aan.